pdf_parsing/parser.py

```diff
@@ -2,102 +2,152 @@
-                        quote_source_found = True
-                        explanation += missing_char
-                    else:
-                        explanation += sentence
-    else:
-        for index, sentence in enumerate(sentences):
-            if index == 0:
-                date = sentence[:-2]
-            elif index == 1:
-                title = sentence
-            else:
-                if not quote_found:
-                    if sentence[-1] != end_quote_symbol:
-                        quote += sentence
-                    else:
-                        quote += sentence
-                        quote_found = True
-                else:
-                    if not quote_source_found:
-                        quote_source = sentence
-                        quote_source_found = True
-                    else:
-                        explanation += sentence
-
-    return date, title, quote, quote_source, explanation
-
-
-def extract_text(pdf_file, start, end, breaks):
-    entries = {}
-
-    with fitz.open(pdf_file) as f:
-        for page in f:
-            if page.number < start:
+PARSING_FOLDER = "./pdf_parsing/"
+RESULT_FILE_PATH = PARSING_FOLDER + "Stoic_log.json"
+
+# These needs to be set according to a specific copy of The Daily Stoic book.
+# Page numbers start from 0, so page number in PDF minus 1
+FIRST_STOIC_PAGE = 14
+LAST_STOIC_PAGE = 392
+SKIP_NON_STOIC_PAGES = [
+    45,
+    75,
+    107,
+    138,
+    139,
+    171,
+    202,
+    234,
+    266,
+    267,
+    298,
+    330,
+    361,
+]
+START_QUOTE_SYMBOL = "\u201c"  # Origin: “
+END_QUOTE_SYMBOL = "\u201d"  # Origin: ”
+START_SOURCE_SYMBOL = "\u2014"  # Origin: —
+
+
+# Parses The Daily Stoic book.
+# The book is a PDF file with a specific structure.
+class Parser:
+    # Returns a list of all pdf files in a folder.
+    def get_pdfs(self, folder="."):
+        return [folder + "/" + file for file in os.listdir(folder) if ".pdf" in file]
+
+    # Parses The Daily Stoic book and returns a dictionary with the entries.
+    def parse_book(self, pdf_file):
+        entries = {}
+
+        with fitz.open(pdf_file) as f:
+            for page in f:
+                if page.number < FIRST_STOIC_PAGE:
+                    continue
+                if page.number > LAST_STOIC_PAGE:
+                    break
+                if page.number in SKIP_NON_STOIC_PAGES:
+                    continue
+
+                date, title, quote, quote_source, explanation = self.__parse_page_text(
+                    page.get_text()
+                )
+
+                entries[date] = {
+                    "title": title,
+                    "quote": quote,
+                    "quote_source": quote_source,
+                    "explanation": explanation,
+                }
+
+        return entries
+
+    # Extracts the date, source, quote and explanation from a page of the Daily Stoic
+    def __parse_page_text(self, text: str):
```

Comment:
I am used to single underscore for 'private' functions/methods, any reason to use __?

Reply:
No reason to use `__` over `_`, beside my own rusty Python skills. 😄
Replaced.